### ISSUE TITLE
Allow overwriting Bun.stdout and Bun.stderr

### ISSUE THREAD

**Issue by TomasHubelbauer** [oven-sh/bun#14874]
### What is the problem this feature would solve?

I have a piece of code where a method writes to the standard output using `console.log` and when a condition is met, it should instead write to a file.

I have no control over the implementation of the method, so I am looking for ways to redirect the `console.log` and `console.error` calls.

I tried doing this:

```
const file = Bun.file('file-name.ext');
Bun.stdout = file;
Bun.stderr = file;

method(); // Calls `console.log` inside
```

However the standard output and standard error streams remained the terminal-attached ones, the file was created, but remained empty.

### What is the feature you are proposing to solve the problem?

Make `Bun.stdout` and `Bun.stderr` read-write and respect the overriden values when using these internally as sinks for `console` methods. (But this should probably also affect `process.stdout` and `process.stderr` maybe?

**Comment by heimskr** [comment]
Would it suffice to replace the `console.log` global?

```js
console.log = (...args) => {
    writeToFile("filename", ...args);
};
```

**Comment by TomasHubelbauer** [comment]
That's what I'm doing right now but it doesn't feel like a correct solution. A way to tell Bun to write to a different `BunFile` whenever stdout or stderr is written into would be preferable.

**Comment by guest271314** [comment]
`console` is not part of ECMA-262. ECMA-262 doesn't specify I/O at all, so whatever happens in JavaScript runtiems or engines, or doesn't happen, is completely arbitrary.

You can't really redirect `console.log`, which is generally a TTY, that I am aware of, you can overwrite `console.log()`. if `console` exists at all in the runtime, which it doesn't have to to be within the language of ECMA-262.

**Comment by guest271314** [comment]
I think you can solve this by making use of the `tee` command.

**Comment by TomasHubelbauer** [comment]
Note that this feature request is about Bun.stdout and Bun.stderr. Console method are just one example of where these sinks are used.

**Comment by guest271314** [comment]
The approach @heimskr suggested would be the simplest.

**Comment by TomasHubelbauer** [comment]
Agreed. That's why I am using it, as I've stated. The feature request still stands.

**Comment by guest271314** [comment]
Then I think you mean overwriting `console`, not `Bun.stdout` and `Bun.stderr`, which really have nothing to do with your situation.

**Comment by guest271314** [comment]
What happens when you continuously read `process.stdout` during running the code you have?
